### ISSUE TITLE
Input device (Adonit Jot Pro) no longer responding properly

### ISSUE THREAD

**Issue by JSDLacoste** [xournalpp/xournalpp#5881]
### Operating System

Linux

### (Linux only) Distribution

Ubuntu

### (Linux only) Desktop Environment

GNOME

### (Linux Only) Display Server

Wayland

### Installation Method

command line

### Xournal++ Version

1.1.1

### libgtk Version

3.24.33

### Bug Description

Up until a few days ago, I have been happily using my Adonit Jot Pro stylus on a Microsoft SurfacePro 5 device running Ubuntu 22.  It is lo longer behaving properly.  While trying to draw a simple line, I get a series of dots.  If I apply extreme pressure, a line shows up as points and dashes.  

### Expected Behaviour

I expect to get a line, not dots and dashes

### Steps to Reproduce

I am just trying to write stuff.  Just simple lines are not working anymore.

### Additional Context

I am really not competent enough to provide the input logs at the moment but I can try to get help from someone if necessary.

**Comment by rolandlo** [comment]
Instead of providing input logs it may help getting `libinput` logs that show stylus events. Specifically:

Run 
```shell
sudo libinput list-devices
```
to see which kernel events (e.g. `/dev/input/event7`) correspond to which device. Then run

```shell
sudo libinput debug-events --device /dev/input/event7
```
(with the device argument corresponding to what you got above) and use the stylus as you would when drawing.
You can write the output into a file (`log.txt`) that you upload here by appending `> log.txt` to the command, i.e.

```shell
sudo libinput debug-events --device /dev/input/event7 > log.txt
```

You may also check if other Gtk 3 apps (like e.g. Inkscape) show the same issue. As a most basic test run the `Paint` example in `gtk3-demo` that can be installed via 
```
sudo apt install gtk-3-examples
```
on Ubuntu.

**Comment by JSDLacoste** [comment]
Hi, libinput command not found.  May I get the commands to install it please?

**Comment by rolandlo** [comment]
I think you only need to install `libinput-tools` via
```term
sudo apt install libinput-tools
```

**Comment by JSDLacoste** [comment]
Thanks.  I successfully installed `libinput-tools` .  Running `sudo libinput list-devices`, I get a list of 10 devices.

**Comment by JSDLacoste** [comment]
Hi, attached is the log.txt corresponding to event5 which activates when I try to use my stylus in xournal++.
[log.txt](https://github.com/user-attachments/files/17107614/log.txt)

**Comment by rolandlo** [comment]
That means your stylus events are received as touch events, which is already quite bad for Xournal++. Moreover it's a combination of many short
```
TOUCH_DOWN
TOUCH_FRAME
TOUCH_UP
```
sequences, like you were constantly putting your finger on the screen and lifting it again. No wonder you get a series of dots.

With those events Xournal++ cannot realistically produce a nice output. It could be a driver issue. Do you have any idea what changed on your system between the time when the stylus worked fine and when it didn't?

**Comment by rolandlo** [comment]
> Yes, that makes sense. As to what changed on my system between the moment it worked and not, all I can say is that I used Update Manager. Also, I use Timeshift to make images of my surface pro. When I reload a Timeshift created on september 13th 2024, the stylus works correctly, but Zotero and Firefox don't. From there, I ran Update Manager, skipping all those related to GTK but I did not recover functionality of Zotero and Firefox. I have screenshots of all those GTK udpdate elements that I have unselected. Would that be useful for you to have them?

It would be interesting to have that list, yes. I'm also curious what libinput logs you get when the stylus is working.

**Comment by JSDLacoste** [comment]
Hi, 

Here are the screenshots.  Log.txt on its way.

![2024-09-22 11 52 42](https://github.com/user-attachments/assets/8e1e89f9-5068-4695-9692-f1c26bd03537)
![2024-09-22 11 51 22](https://github.com/user-attachments/assets/46095d40-0e04-4466-ac83-b0c1e674bddb)
![2024-09-22 11 44 25](https://github.com/user-attachments/assets/c9a69a82-451d-4d7e-bf53-99dfc1362c61)
![2024-09-22 11 44 10](https://github.com/user-attachments/assets/4d18fdc7-dc0b-4df3-8708-78ff9eb18090)
![2024-09-22 11 43 58](https://github.com/user-attachments/assets/0686c0b4-947e-4a7d-a5a9-6b93205f45ab)
![2024-09-22 11 43 35](https://github.com/user-attachments/assets/d360b72c-b394-4806-ad22-1e9d28a8d5ab)
![2024-09-22 11 43 24](https://github.com/user-attachments/assets/44f5262c-4cb5-40c6-a52f-56a145445b18)
![2024-09-22 11 40 36](https://github.com/user-attachments/assets/c7f30b71-88e5-4030-b6db-857609da0fcf)
![2024-09-22 11 35 36](https://github.com/user-attachments/assets/66a0c122-1f96-4435-be8a-7001be00cf8e)

**Comment by JSDLacoste** [comment]
Hi, 
When my stylus is working correctly, the listed input devices have two elements not showing up when my stylus is not working correctly:
IPTS Touch (event 19)
IPTS Stylus (event 18)

Also, for both of the above, running the command `sudo libinput debug-events --device /dev/input/event7 > log.txt` for both of the above returns the error message `Failed to initialize device.

Best regards,

Judith

**Comment by rolandlo** [comment]
From the unselected non-kernel related packages I see:
- `libwebkitgtk` (web integration for Gtk 3) -> could only be relevant for Firefox
- `libvte` (terminal emulator widget for Gtk 3) -> irrelevant
- `libgtk-3-bin` (helper programs for Gtk 3) -> irrelevant for the stylus
- `gtk2-engines-pixbuf` (engine for pixbuf themes) ->  irrelevant for the stylus
- `libreoffice-gtk3` (Gtk 3 integration in LibreOffice) -> irrelevant
- `libgtk2.0.0` (Gtk 2) and corresponding gintrospection files -> irrelevant

Not on the list: `libgtk-3-0`, the `GTK graphical user interface library`. 

Could it be that the stylus working or not only has to do with the kernel upgrade?

**Comment by JSDLacoste** [comment]
Hi,
I keep reading your reply and I am afraid I don't understand it.  Do you mean I should install libgtk-3-0?

**Comment by rolandlo** [comment]
> Hi, I keep reading your reply and I am afraid I don't understand it. Do you mean I should install libgtk-3-0?

I would try to upgrade all the Gtk related packages without upgrading the kernel.

**Comment by JSDLacoste** [comment]
Hi,
I was able to brainstorm with someone more competent with a similar set up.  Here is what we found.  In the Ubuntu Software Updater, in a section named "Other updates", there is a file named "Userspace daemon for Intel Precise Touch & Stylus".  My issue comes up as soon as this file is updated.  So now I am careful to leave it unchecked.  It turns out my issue with Firefox was something completely unrelated. 

I am grateful for your work.  Xournal++ is fantastic!  

Best regards,

Judith
